### PR TITLE
Turned most public `List<T>`-typed properties into `Collection<T>` or `ReadOnlyCollection<T>`

### PR DESCRIPTION
**Pull request by Lehonti**
There are three things a public `List<T>` means:
- The items exposed through the collection are set in stone, in which case a `ReadOnlyCollection<T>` or other read-only collection type should be used. I didn't find any instances of this, though.
- It holds some part of the internal state of the object, and it shouldn't be modified externally, but only through the appropriate methods in the class interface. In this case, the backing `List<T>` was made private, and the property was turned into a `ReadOnlyCollection<T>` that wraps the backing `List<T>`
- You want to allow the user to add/remove items, so that something can be configured. The problem with `List<T>` is that it's too "raw". It's better to expose a `Collection<T>`, as its methods `ClearItems`, `InsertItem`, `RemoveItem`, and `SetItem` are `virtual`, allowing us to change its implementation without breaking the public interface. Imagine if at some point you want to start validating the items that an external caller is trying to insert, for example.

As an extra, I `sealed` some classes and modernized some getter/setter syntax as I found it

### CODE REVIEW

**Comment by cameronwhite** [comment]
There are a few compiler warnings introduced (`new protected member declared in sealed type`)

**Comment by Lehonti** [comment]
> There are a few compiler warnings introduced (`new protected member declared in sealed type`)

I made the methods `private`. If the classes were (for some reason) be un-`sealed` in the future, the methods would have to be made `protected` and (perhaps most importantly) `virtual` according to need.

File: Pinta.Core/Managers/WorkspaceManager.cs [hunk 1]
```diff
@@ -44,66 +45,63 @@ public interface IWorkspaceService
...
-	public class WorkspaceManager : IWorkspaceService
+	public sealed class WorkspaceManager : IWorkspaceService
 	{
 		private int active_document_index = -1;
 		private int new_file_name = 1;
 
 		public WorkspaceManager ()
 		{
-			OpenDocuments = new List<Document> ();
+			var openDocumentsBacking = new List<Document> ();
```
Comment (cameronwhite) [inline Pinta.Core/Managers/WorkspaceManager.cs:55]: I feel like this is adding more complexity for someone reading the code. Can we remove the temporary `openDocumentsBacking` and rename `open_documents_backing` to just `open_documents`?
(Similar for the other classes that were modified)

Comment (Lehonti) [reply Pinta.Core/Managers/WorkspaceManager.cs]: Yes, it's those pesky line endings. I've now reduced the annoyance to the minimum, though, only one line that for some reason was not normalized remains

**Pull request merged**